src/modules/BootStorage.rb

```diff
@@ -213,36 +111,123 @@ def available_swap_partitions
     end
 
     def encrypted_boot?
-      dev = boot_partition
-      log.info "boot device = #{dev.inspect}"
-      # check if on physical partition is any encryption
-      result = dev.descendants.any? { |a| a.is?(:encryption) }
+      fs = boot_mountpoint
+      log.info "boot mp = #{fs.inspect}"
+      # check if fs is on an encryption
+      result = fs.ancestors.any? { |a| a.is?(:encryption) }
 
       log.info "encrypted_boot? = #{result}"
 
       result
     end
 
+    # get suitable device for stage1 by string name
+    # @param [String] dev_name device name
+    # @return [Array<Y2Storage::Device>] list of suitable devices
+    def stage1_device_for_name(dev_name)
+      device = staging.find_by_name(dev_name)
+      raise "unknown device #{dev_name}" unless device
+
+      if device.is?(:partition) || device.is?(:filesystem)
+        stage1_partitions_for(device)
+      else
+        stage1_disks_for(device)
+      end
+    end
+
+    # get stage1 device suitable for stage1 location
+    # ( so e.g. exclude logical partition and use instead extended ones )
+    # @param [Y2Storage::Device] device to check
+    # @return [Array<Y2Storage::Device] devices suitable for stage1
+    def stage1_partitions_for(device)
+      # so how to do search? at first find first partition with parents
+      # that is on disk or multipath (as ancestors method is not sorted)
+      to_process = [device]
+      partitions = []
+      loop do
+        break if to_process.empty?
+        partition = to_process.pop
+        if partition.is?(:partition)
+          partitionable = partition.partitionable
+          if partitionable.is?(:disk) || partitionable.is?(:multipath)
+            partitions << partition
+            next # we are done here, we found partition for this part
+          end
+        end
+        to_process.concat(partition.parents)
+      end
+
+      # now replace all logical partitions for extended
+      partitions.map! { |p| extended_for_logical(p) }
+      partitions.uniq!
+
+      log.info "stage1 partitions for #{device.inspect} is #{partitions.inspect}"
+
+      partitions
+    end
+
+    # If passed partition is logical, return extended, otherwise return argument
+    def extended_for_logical(partition)
```

Comment:
Not covered by tests, and neither is `extended_partition`.

Reply:
I will check if I have proper storage data with logical partition.

Reply:
test added